source/common/router/router.cc

```diff
@@ -520,6 +520,21 @@ void Filter::onUpstreamHeaders(Http::HeaderMapPtr&& headers, bool end_stream) {
       upstream_request_->upstream_host_->canary();
   chargeUpstreamCode(*headers, upstream_request_->upstream_host_);
 
+  // Append routing cookies
+  for (auto& cookie : downstream_set_cookies_) {
+    const std::string& key = std::get<0>(cookie);
+    const std::string& value = std::get<1>(cookie);
+    const int max_age = std::get<2>(cookie);
+
+    // Don't overwrite a cookie if it's already set by the upstream host
+    if (Http::Utility::hasSetCookie(*headers, key)) {
```

Comment:
Do we think there's a need for this year?  It seems unlikely that the upstream host would be able to set a cookie which would result in hashing back to the same host.